modules/xfeatures2d/src/daisy.cpp

```diff
@@ -0,0 +1,1754 @@
+    for( int x=-sz; x<=sz; x++ )
+    {
+      counter++;
+      fltr[counter] = exp((-((float)x-mean)*((float)x-mean))/v);
+      sum += fltr[counter];
+    }
+
+    if( sum != 0 )
+    for( int x=0; x<fsz; x++ ) fltr[x] /= sum;
+}
+
+// computes the gradient of an image
+static void gradient( Mat im, int h, int w, Mat dy, Mat dx )
+{
+    CV_Assert( !dx.empty() );
+    CV_Assert( !dy.empty() );
+
+    for( int y=0; y<h; y++ )
+    {
+      int yw = y*w;
+      for( int x=0; x<w; x++ )
+      {
+        int ind = yw+x;
+        // dx
+        if( x>0 && x<w-1 ) dx.at<float>(ind) = (im.at<float>(ind+1)-im.at<float>(ind-1)) / 2.0f;
+        if( x==0         ) dx.at<float>(ind) =  im.at<float>(ind+1)-im.at<float>(ind  );
+        if( x==w-1       ) dx.at<float>(ind) =  im.at<float>(ind  )-im.at<float>(ind-1);
+
+        // dy
+        if( y>0 && y<h-1 ) dy.at<float>(ind) = (im.at<float>(ind+w)-im.at<float>(ind-w)) / 2.0f;
+        if( y==0         ) dy.at<float>(ind) =  im.at<float>(ind+w)-im.at<float>(ind  );
+        if( y==h-1       ) dy.at<float>(ind) =  im.at<float>(ind  )-im.at<float>(ind-w);
+      }
+    }
+}
+
+static Mat layered_gradient( Mat data, int layer_no = 8 )
+{
+   int data_size = data.rows * data.cols;
+   Mat layers( 1, layer_no*data_size, CV_32F, Scalar(0) );
+
+   float kernel[5];
+   gaussian_1d(kernel, 5, 0.5f, 0.0f);
+   Mat Kernel(1, 5, CV_32F, (float*) kernel);
+
+   Mat cvO;
+   // smooth the data matrix
+   filter2D( data, cvO, CV_32F, Kernel, Point( -1, -1 ), 0, BORDER_REPLICATE );
+   filter2D( cvO, cvO, CV_32F, Kernel.t(), Point( -1, -1 ), 0, BORDER_REPLICATE );
+
+   Mat dx(1, data_size, CV_32F);
+   Mat dy(1, data_size, CV_32F);
+
+   gradient(cvO, data.rows, data.cols, dy, dx);
+   cvO.release();
+
+#if defined _OPENMP
+#pragma omp parallel for
+#endif
+   for( int l=0; l<layer_no; l++ )
+   {
+      float angle = (float) (2*l*CV_PI/layer_no);
+      float kos = (float) cos( angle );
+      float zin = (float) sin( angle );
+
+      float* layer_l = layers.ptr<float>(0) + l*data_size;
+
+      for( int index=0; index<data_size; index++ )
+      {
+         float value = kos * dx.at<float>(index) + zin * dy.at<float>(index);
+         if( value > 0 ) layer_l[index] = value;
+         else            layer_l[index] = 0;
+      }
+   }
+   return layers;
+}
+
+// data is not destroyed afterwards
+static void layered_gradient( Mat data, int layer_no, Mat layers )
+{
+   CV_Assert( !layers.empty() );
+
+   Mat cvI = data.clone();
+   layers.setTo( Scalar(0) );
+   int data_size = data.rows * data.cols;
+
+   float kernel[5];
+   gaussian_1d(kernel, 5, 0.5f, 0.0f);
+   Mat Kernel(1, 5, CV_32F, (float*) kernel);
+
+   filter2D( cvI, cvI, CV_32F, Kernel, Point( -1, -1 ), 0, BORDER_REPLICATE );
+   filter2D( cvI, cvI, CV_32F, Kernel.t(), Point( -1, -1 ), 0, BORDER_REPLICATE );
+
+   Mat dx(1, data_size, CV_32F);
+   Mat dy(1, data_size, CV_32F);
+   gradient( cvI, data.rows, data.cols, dy, dx );
```

Comment:
I'll try. Only if you confirm that would be acceptable to have slight different numeric results (but obviously same quality) as original daisy.